test/jdk/java/net/URLConnection/B5052093.java

```diff
@@ -89,7 +97,20 @@ public static void main(String[] args) throws Exception {
                 throw new RuntimeException("Wrong content-length from file");
             }
         } finally {
-            server.terminate();
+            server.stop(1);
+        }
+    }
+}
+
+class B5052093Handler implements HttpHandler {
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+        try {
+            exchange.getResponseHeaders().set("content-length", Long.toString(B5052093.testSize));
+            exchange.sendResponseHeaders(200, 0);
+            exchange.close();
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Are you sure that this results in the same response headers than before?
If I'm not mistaken here we will send both Content-Length and Encoding: chunked. Was that what the previous server did, and what the test wants to test?

Reply:
Yes, previously also, setting the content-length Integer.MAX_VALUE)) + 2, and was not sending any content. Here wanted to test, that URLConnection.getContentLength() does not throw NumberFormatException and return -1 if content-length is long value.
In case of HttpExchange.setResponseHeader(). If responseLength is -1, then content-length value is overridden to 0, if already set explicitly. Same is the case when responseLength is > 0. Only in the case when responseLength == 0, content-length value is not overriden if already set explicitly., thats why I am using chunked encoding and not writing any data.